### PR TITLE
Fixed #14529 by disposing all ever created cells by of a listview in …

### PR DESCRIPTION
**Pull request by gentledepp**
…the UWP ListViewRenderer.Dispose method

### Issues Resolved ### 

- fixes #14529
- https://github.com/xamarin/Xamarin.Forms/issues/14529

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Behavioral/Visual Changes ###

- UWP

### CODE REVIEW

File: Xamarin.Forms.Core/ListView.cs [hunk 1]
```diff
@@ -644,5 +644,25 @@ protected override bool ValidateItemTemplate(DataTemplate template)
 
 			return true;
 		}
+		
+		public IEnumerable<Element> ListViewLogicalChildren => _logicalChildren.AsEnumerable();
+
+		public void AddLogicalChild(Cell content)
```
Comment (bruzkovsky) [inline Xamarin.Forms.Core/ListView.cs]: the parameter should be called `cell`, imho

File: Xamarin.Forms.Platform.UAP/CellControl.cs [hunk 2]
```diff
@@ -388,7 +388,8 @@ void SetCell(object newContext)
 
 				// A TableView cell should already have its parent,
 				// but we need to set the parent for a ListView cell.
-				cell.Parent = lv;
+				//cell.Parent = lv;
```
Comment (bruzkovsky) [inline Xamarin.Forms.Platform.UAP/CellControl.cs]: please remove this comment

File: Xamarin.Forms.Core/ListView.cs [hunk 3]
```diff
@@ -644,5 +644,25 @@ protected override bool ValidateItemTemplate(DataTemplate template)
...
+		{
+
+			if (content != null)
+				_logicalChildren.Add(content);
+
+			content.Parent = this;
+			VisualDiagnostics.OnChildAdded(this, content);
```
Comment (bruzkovsky) [inline Xamarin.Forms.Core/ListView.cs]: @gentledepp what does this line (657) do? Never seen it 😲

File: Xamarin.Forms.Platform.UAP/ListViewRenderer.cs [hunk 4]
```diff
@@ -326,6 +327,29 @@ protected override void Dispose(bool disposing)
 					List = null;
 				}
 
+				if (Element != null)
```
Comment (bruzkovsky) [inline Xamarin.Forms.Platform.UAP/ListViewRenderer.cs:330]: how do the other platforms handle this? Should we do this on Android/iOS too?

File: Xamarin.Forms.Core/ListView.cs (hunk 3 above)
Comment (gentledepp) [reply Xamarin.Forms.Core/ListView.cs]: which one? "VisualDisgnostics"?
I have no idea - this is essentially copied code from `ListView.SetupContent`  and `ListView.UnhookContent`
which is also why the `Cell` parameter is called "content" ;-)

see here:
```csharp



		protected override void SetupContent(Cell content, int index)
		{
			base.SetupContent(content, index);
			if (content is ViewCell viewCell && viewCell.View != null && HasUnevenRows)
				viewCell.View.ComputedConstraint = LayoutConstraint.None;

			if (content != null)
				_logicalChildren.Add(content);

			content.Parent = this;
			VisualDiagnostics.OnChildAdded(this, content);
		}

		protected override void UnhookContent(Cell content)
		{
			base.UnhookContent(content);

			if (content == null || !_logicalChildren.Contains(content))
				return;
			var index = _logicalChildren.IndexOf(content);
			_logicalChildren.Remove(content);
			content.Parent = null;
			VisualDiagnostics.OnChildRemoved(this, content, index);

		}


```

File: Xamarin.Forms.Platform.UAP/ListViewRenderer.cs (hunk 4 above)
Comment (gentledepp) [reply Xamarin.Forms.Platform.UAP/ListViewRenderer.cs:330]: Nope. On Android, the `ListViewAdapter` keeps track of **all** created views in its field
```readonly List<ConditionalFocusLayout> _layoutsCreated = new List<ConditionalFocusLayout>();```

I did not check iOS, but I guess this works as well - I do not have the necessary tooling to check this anyways.

This seems to be a UWP bug only

File: Xamarin.Forms.Core/ListView.cs (hunk 3 above)
Comment (bruzkovsky) [reply Xamarin.Forms.Core/ListView.cs]: Ok, so be it :)

**Review by bruzkovsky — APPROVED** [review]

**Review by eszterhorvath — APPROVED** [review]

**Pull request merged**